### ISSUE TITLE
Swagger UI always shows up if path is used(partial, eg. as base path)

### ISSUE THREAD

**Issue by mathis-m** [nestjs/swagger#270]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

If the swagger path is set to a used path, it will show up always.

## Expected behavior

It should be possible to set the path to a path that may be partially included in a route's path.

## Minimal reproduction of the problem with instructions

```typescript
async function bootstrap() {
	const app = await NestFactory.create(AppModule);
	app.setGlobalPrefix('v1');

	const options = new DocumentBuilder()
		.setBasePath('v1')
		.build();
	const document = SwaggerModule.createDocument(app, options);
	SwaggerModule.setup('v1', app, document);

	await app.listen(5050);
}
```
Navigate to a defined route => Swagger will show up.

## What is the motivation / use case for changing the behavior?

If I don't know about the structure of an endpoint, I also don't know the path for swagger ui, I want to be able to navigate to the base path to see swagger docs. 

## Environment

<pre><code>
Nest version: 
"@nestjs/common": "^6.3.1",
"@nestjs/core": "^6.3.1",
"@nestjs/platform-express": "^6.3.1",
"@nestjs/swagger": "^3.0.2",

 
For Tooling issues:
- Node version: v12.4.0
- Platform:  Windows

Others:

OS Name:                   Microsoft Windows 10 Enterprise
OS Version:                10.0.14393 N/A Build 14393
IDE: 
WebStorm 2019.1.3
Build #WS-191.7479.14, built on May 27, 2019
npm: 6.9.0
</code>
</pre>

**Comment by mathis-m** [comment]
Possible workarround:
```typescript

const setupExpress = async (path, app, document, options) => {
	const validatePath = (p) => p.charAt(0) !== '/' ? '/' + p : p;
	const finalPath = validatePath(path);
	const swaggerUi = await import('swagger-ui-express');
	const swaggerHtml = swaggerUi.generateHTML(document, options);
	app.use(finalPath, swaggerUi.serveFiles(document, options));
	app.use(finalPath, (req: Request, res, next) => {
		if (req.url !== '/') {
			return next();
		}
		return res.send(swaggerHtml);
	});
	app.use(finalPath + '-json', (req, res) => res.json(document));
};

async function bootstrap() {
	const app = await NestFactory.create(AppModule);
	app.setGlobalPrefix('v1');

	const options = new DocumentBuilder()
		.setBasePath('v1')
		.build();
	const document = SwaggerModule.createDocument(app, options);

	try {
		await setupExpress('v1', app, document, options);
		await app.listen(5050);
	}
	catch (e) {
		throw new Error(e);
	}
}

bootstrap();

```

**Comment by kamilmysliwiec** [comment]
Fixed in 3.1.0
